JavaFXUserInterface/src/main/java/Controllers/AssetTypeInfoController.java

```diff
@@ -495,4 +555,158 @@ public boolean formInputValidation() {
+     */
+    private void highlightAssociatedModel(ObservableList<Pane> modelPanes) {
+        int associatedModelID = modelDAO.getModelIDFromAssetTypeID(assetType.getId());
+        for (int i = 0; i < modelPanes.size(); i++) {
+            if ((i + 1) == associatedModelID) {
+                modelPanes.get(i).setStyle("-fx-border-color: red");
+            }
+            else if((i + 1) != associatedModelID) {
+                modelPanes.get(i).setStyle("-fx-background-color: #e0e0eb");
+            }
+        }
+    }
+
+    /**
+     * This function properly sets up the flow pane to scroll horizontally only and other
+     * settings
+     *
+     * @param modelPanes is the observable list of panes that contain model information
+     * @author Jeremie
+     */
+    private void setModelThumbnailsContainerPane(ObservableList<Pane> modelPanes) {
+        modelsThumbPane.setPrefWidth((300.0 + modelsThumbPane.getHgap()) * (modelPanes.size()) + (2 * modelsThumbPane.getHgap()));
+        modelsThumbPane.getChildren().addAll(modelPanes);
+        modelsThumbPane.setOnMouseClicked(mouseEvent -> handleModelSelectionChange(modelPanes));
+    }
+
+    /**
+     * This function handles the UI display of changing the user selection from one model
+     * pane to another.
+     *
+     * @param modelPanes is the observable list of panes that contain model information
+     * @author Jeremie
+     */
+    private void handleModelSelectionChange(ObservableList<Pane> modelPanes) {
+        for (int i = 0; i < modelPanes.size(); i++) {
+            if ((i + 1) != selectedModelIndex) {
+                modelPanes.get(i).setStyle("-fx-border-color: transparent; -fx-background-color: #e0e0eb");
+            }
+        }
+    }
+
+    /**
+     * This function handles the selection of a model.
+     *
+     * @param model is the selected model
+     * @param pane is the pane containing the selected model
+     * @author Jeremie
+     */
+    private void handleModelSelection(Model model, Pane pane) {
+        pane.setStyle("-fx-border-color: red");
+        selectedModel = model;
+        selectedModelIndex = Integer.parseInt(model.getModelID());
+        modelSaveBtn.setDisable(false);
+    }
+
+    private void updateThumbnails() {
+        modelsThumbPane.getChildren().removeAll();
+        generateThumbnails();
+    }
+
+    /**
+     * Saves the association between the selected model and the current asset type in the database.
+     * This function also set the retrain tag to true to indicate that the model chosen for the asset
+     * type needs to be retrain. Finally, it labels all live assets of that asset type as updated.
+     *
+     * @author Jeremie
+     */
+    private void saveSelectedModelAssociation() {
+        modelDAO.updateModelAssociatedWithAssetType(selectedModel.getModelID(), assetType.getId());
+        modelDAO.setModelToTrain(assetType.getId());
+        updateAssetsForSelectedModelAssociation(assetType.getId());
+    }
```

Comment:
Issue regarding the function called in modelSaveBtn. The database is not updated when the save button is clicked once another model is selected.

Reply:
@Shirwarsame @t98b The reason why the selection was not saved is because the save button could not be clicked in the first place. The calls attached to the save button were good, but the button could never be clicked because there was another UI element on top of it that would prevent the clicking. The element was a transparent anchor pane. I removed it in my latest commit. Sorry for that.